content/docs/cli/_index.md

```diff
@@ -93,10 +93,12 @@ Add one or more k8s clusters to be managed by dellctl
 ##### Flags
 
 ```
+Flags:
+  -n, --names strings   cluster names
   -f, --files strings   paths for kube config files
+  -u, --uids strings    uids of the kube-system namespaces in the clusters
```

Comment:
Can we indicate all the flags that are optional ?

Reply:
We use cobra module for cli development. The same is used by kubectl, velero etc.
The help message is auto generated by the cobra module. Currently it doesn't mark any of the flags as required/optional, in the help message, even though the flag is marked as required in the code. But it errors out if the mandatory flag is not specified on the cli.

For example, the flag "names" is marked as required in the code:
```
	_ = clusterAddCmd.Flags().StringSliceP("names", "n", []string{}, "cluster names")
	_ = clusterAddCmd.MarkFlagRequired("names")

```
When the command is run without this flag:

```
 -> dellctl cluster add
Error: required flag(s) "names", "files" not set
Usage:
  dellctl cluster add [flags]

Examples:
  dellctl cluster add -n <cluster-name> -f <config-file>
  dellctl cluster add -n <cluster-name1> -f <config-file1> -n <cluster-name2> -f <config-file2> ...
  dellctl cluster add -n <cluster-name1,cluster-name2,...> -f <config-file1,config-file1,...>
  dellctl cluster add -n <cluster-name> -f <config-file> -u <cluster-uid>


Flags:
  -n, --names strings   cluster names
  -f, --files strings   paths for kube config files
  -u, --uids strings    uids of the kube-system namespaces in the clusters
      --force           forcefully add cluster
  -h, --help            help for add

required flag(s) "names", "files" not set

```

To overcome this issue we can have custom help messages for each of the commands. We can explore and consider this during the next release.

Reply:
Ok sounds good. Thanks.